tests/unit/test_run.py

```diff
@@ -0,0 +1,36 @@
+import pytest
+
+from dvc.repo.run import is_valid_name, parse_params
+
+
+def test_parse_params():
+    assert parse_params(
+        [
+            "param1",
+            "file1:param1,param2",
+            "file2:param2",
+            "file1:param2,param3,",
+            "param1,param2",
+            "param3,",
+            "file3:",
+        ]
+    ) == [
+        "param1",
+        {"file1": ["param1", "param2"]},
+        {"file2": ["param2"]},
+        {"file1": ["param2", "param3"]},
+        "param1",
+        "param2",
+        "param3",
+        {"file3": []},
+    ]
+
+
+@pytest.mark.parametrize("name", ["copy_name", "copy-name", "copy-name", "12"])
```

Comment:
why do we repeat copy-name here?

Reply:
Thanks @pared. It should have been plain `copyname` (i improvised it to `copyName` now, you know just in case if I had missed uppercase letters to whitelist)